pyqtgraph/examples/ExampleApp.py

```diff
@@ -369,7 +369,20 @@ def filterByTitle(self, text):
         self.hl.setDocument(self.ui.codeView.document())
 
     def filterByContent(self, text=None):
-        # Don't filter very short strings
+        # If the new text isn't valid regex, fail early and highlight the search filter red to indicate a problem
+        # to the user
+        validRegex = True
+        try:
+            re.compile(text)
+            background = app.palette().background().color()
+        except re.error:
+            colors = DarkThemeColors if app.property('darkMode') else LightThemeColors
+            background = pg.mkColor(colors.Red)
+            background.setAlpha(100)
+            validRegex = False
+        self.ui.exampleFilter.setStyleSheet(f'background: {pg.mkColor(background).name(QtGui.QColor.HexArgb)}')
```

Comment:
QtGui.QColor.NameFormat.HexArgb?

Reply:
Does it work to just set 
``` python
try:
    re.compile(text)
    self.ui.exampleFilter.setStyleSheet('')
except:
    colors = DarkThemeColors if app.property('darkMode') else LightThemeColors
    qcol = QtGui.QColor(colors.Red)
    qcol.setAlpha(100)
    hexcol = qcol.name(QtGui.QColor.NameFormat.HexArgb)
    self.ui.exampleFilter.setStyleSheet(f'background: {hexcol})
    return
``` 
?
Going through mkColor twice just to prefix #64 to the string seems like the long way around.

And I think clearing the stylesheet should revert to the original colors, without explicitly setting one. Did not test, though.

Reply:
@NilsNemitz beat me to it, but yeah, get the `colors` via accessing `DarkThemeColors` or `LightThemeColors` based on the `darkMode` property.

Since the colors come in string form, I don't think we should even convert to qcolor.

```python
colors = DarkThemeColors if app.property('darkMode') else LightThemeColors
red = colors.Red
self.ui.exampleFilter.setStyleSheet(f'background : {red}00')
```

Reply:
Qt's stylesheet seems to be AARRGGBB, though. Not sure if 
`self.ui.exampleFilter.setStyleSheet(f'background : #64{red[1:]}')` 
is more readable than the detour through QColor.  
:)

Reply:
oh yeah, `rgba()` is likely far more readable

```python
colors = DarkThemeColors if app.property('darkMode') else LightThemeColors
error_color = pg.mkColor(colors.Red)
self.ui.exampleFilter.setStyleSheet(f'background : rgba({error_color.red()}, {error_color.green()}, {error_color.blue(), {error_color.alpha()})
```